### PR TITLE
Improvements that include incompatible changes (catalyst 3.0)

### PR DESCRIPTION
**Pull request by douglasbagnall**
These ones go on top of #30, and are essentially the same as #28.

These ones include changes to the default behaviour of the web server (not listening to everyone), and to the on-disk save format.

### CODE REVIEW

**Comment by douglasbagnall** [comment]
benchmark results, generated using `./test/time-commits` on an 8-core i7-4770. 

**baseline (ab22e07)**
```
score                0.876
precision            0.876
recall               0.877
score_deviation      0.00464
accuracy             0.877
duration:          508.08
```
**float 32,  no invariant columns (6c4f882)**
```
score                0.886
precision            0.886
recall               0.886
score_deviation      0.00538
accuracy             0.886
duration:           99.41
```
**keras, with one hidden layer (628f98d)**
```
score                0.918
precision            0.921
recall               0.917
score_deviation      0.00262
accuracy             0.917
duration:           67.46
```
**head (f10140b)**
```
score                0.921
precision            0.922
recall               0.922
score_deviation      0.00232
accuracy             0.922
duration:           70.28
```

**Comment by douglasbagnall** [comment]
A couple of notes about this benchmark (based on anonymised real data in `tests/test-requests/test-415-37953-training.bz2`):

1. the correct answer is '0' 84% of the time. In other words, you can get an accuracy of 0.84 by always saying '0'. That is important context when looking at the observed accuracy in the range 0.87 - 0.92, but it does not mean these classifiers are bad. It is a risky thing for the classifier to say '1', and to actually get that right often enough to improve on 84% is a notable feat.

2. The `accuracy` and `recall` numbers are suspiciously similar. I think there is a bug somewhere, and it seems to be in our use of the sklearn library.

**Comment by douglasbagnall** [comment]
Note: the tests do not exserice import and export, and this branch in particular is likely to break that.

(import of models from older versions is bound to fail in any case, but that is true of many releases (e.g. 2.4 which changed the size of the hidden layer). 

Also non-binary classification is not tested.
